awx/api/serializers.py

```diff
@@ -5474,6 +5475,32 @@ def get_percent_capacity_remaining(self, obj):
     def get_health_check_pending(self, obj):
         return obj.health_check_pending
 
+    def peers_read_only(self, attrs, value):
+        # `peers` field in API should be read-only for control type nodes
+        def get_field_from_model_or_attrs(fd):
+            return attrs.get(fd, self.instance and getattr(self.instance, fd) or None)
+
+        node_type = get_field_from_model_or_attrs("node_type")
+        if node_type == 'control' 'peers'(read_only=True):
```

Comment:
looks strange, the peers(readonly) should go on next line after colon

Reply:
Thank you so much! Still working on this one, but I absolutely appreciate your feedback!